### ISSUE TITLE
[ERROR] TypeError: trackTypeDependency is not a function

### ISSUE THREAD

**Issue by racerxdl** [Shopify/graphql-js-client#88]
Hi, I was trying to make a very simple client (like the one in the example) but when running the query I get this error:

```
TypeError: trackTypeDependency is not a function
    at new SelectionSet (/home/lucas/Works/abc/abc-signup/backend/node_modules/graphql-js-client/index.js:644:5)
    at Query.Operation (/home/lucas/Works/abc/abc-signup/backend/node_modules/graphql-js-client/src/operation.js:63:27)
    at new Query (/home/lucas/Works/abc/abc-signup/backend/node_modules/graphql-js-client/index.js:1061:106)
    at Client.query (/home/lucas/Works/abc/abc-signup/backend/node_modules/graphql-js-client/index.js:1802:14)
    at abcIFace._callee$ (/home/lucas/Works/abc/abc-signup/backend/src/services/qiface.js:37:34)
    at tryCatch (/home/lucas/Works/abc/abc-signup/backend/node_modules/regenerator-runtime/runtime.js:64:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/home/lucas/Works/abc/abc-signup/backend/node_modules/regenerator-runtime/runtime.js:299:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/home/lucas/Works/abc/abc-signup/backend/node_modules/regenerator-runtime/runtime.js:116:21)
    at step (/home/lucas/Works/abc/abc-signup/backend/src/services/qiface.js:28:191)
    at /home/lucas/Works/abc/abc-signup/backend/src/services/qiface.js:28:437
    at abcIFace.<anonymous> (/home/lucas/Works/abc/abc-signup/backend/src/services/qiface.js:28:99)
    at abcIFace.DataByCPF (/home/lucas/Works/abc/abc-signup/backend/src/services/qiface.js:112:21)
    at _callee$ (/home/lucas/Works/abc/abc-signup/backend/src/app.js:109:16)
    at tryCatch (/home/lucas/Works/abc/abc-signup/backend/node_modules/regenerator-runtime/runtime.js:64:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/home/lucas/Works/abc/abc-signup/backend/node_modules/regenerator-runtime/runtime.js:299:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/home/lucas/Works/abc/abc-signup/backend/node_modules/regenerator-runtime/runtime.js:116:21)
```

Is that a bug or something I did wrong?

Here is my module:

```javascript
/**
 * Created by Lucas Teske on 08/05/17.
 * @flow
 */

import GraphQLClient from 'graphql-js-client';
import {
  abcColors,
} from 'abc-commons';

import types from './qiface/types';
import {
  iFaceURL,
} from '../config';

abcColors();

export default class abcIFace {

  _client: any;

  constructor() {
    // $FlowFixMe
    console.log('QIFace -- Initializing GraphQL Client to abc Translation Interface'.info);
    this._client = new GraphQLClient(types, {
      url: iFaceURL,
      fetcherOptions: {
        headers: 'X-Powered-By: abc Signup',
      },
    });
  }

  async DataByCPF(cpf: string) {
    // $FlowFixMe
    console.log(`QIFace -- DataByCPF(${cpf.warn.bold})`.debug);
    try {
      const query = this._client.query((root) => {
        root.add('DataByCPF', { args: { cpf } }, (dataByCPF) => {
          dataByCPF.add('name');
          dataByCPF.add('phone');
          dataByCPF.add('addressAddress');
          dataByCPF.add('addressNumber');
          dataByCPF.add('addressComplement');
          dataByCPF.add('addressDistrict');
          dataByCPF.add('addressPostalCode');
          dataByCPF.add('addressCity');
          dataByCPF.add('addressState');
          dataByCPF.add('cpf');
          dataByCPF.add('gender');
          dataByCPF.add('mother');
          dataByCPF.add('birthdate');
        });
      });

      const objects = await this._client.send(query);
      console.log(objects);
      return objects;
    } catch (e) {
      // $FlowFixMe
      console.log(`QIFace -- DataByCPF(${cpf.warn.bold}) Error: ${e.toString().warn.bold}`.error);
      console.log(e);
      return null;
    }
  }
}
```

**Comment by racerxdl** [comment]
I had to remove all entries of `trackFieldDependency` and `trackTypeDependency` on your `index.js`. Even though it was defined as var on the header:
```
var trackTypeDependency = noop.trackTypeDependency;
var trackFieldDependency = noop.trackFieldDependency;
```

Maybe an ES6 related issue?

**Comment by minasmart** [comment]
Yep. I totally broke this at `0.5.1`. In the short term, `0.5.0` should work just fine, and the only missing functionality is around profiling / dependency tracking. 

This happened because in prod builds, I'm stripping out certain functions, and pointing them at `noop`, however, if the functions I'm targeting cease being the primary export of the module, they compile to `noop.someFunction` which doesn't exist. I need a better way to strip dev dependencies, and will hopefully have a fix out later today. 

Thanks for bringing this up!

**Comment by racerxdl** [comment]
Sure, no problem. That happens, I though it was something like that.
Anyways, I will switch over `0.5.0`  for now and wait for a new release. Thanks!

**Comment by minasmart** [comment]
Should be fixed in #89. I'll be pushing that out as `v0.6.1` once I get a couple reviews.

**Comment by minasmart** [comment]
Alright, a fix should be out in `v0.6.1`. Please re-open if this is still for some reason broken.
